pilot/pkg/config/kube/gateway/testdata/mesh.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: networking.x-k8s.io/v1alpha1
+kind: GatewayClass
+metadata:
+  name: istio
+spec:
+  controller: istio.io/gateway-controller
+---
+apiVersion: networking.x-k8s.io/v1alpha1
+kind: Gateway
+metadata:
+  name: gateway
+  namespace: default
+spec:
+  gatewayClassName: istio
+  listeners:
+  - hostname: "*.example.com"
+    port: 80
+    protocol: HTTP
+    routes:
+      kind: HTTPRoute
+---
+apiVersion: networking.x-k8s.io/v1alpha1
+kind: HTTPRoute
+metadata:
+  name: echo
+  namespace: default
+spec:
+  gateways:
+    allow: FromList
+    gatewayRefs:
+      - name: experimental.istio.io/mesh
```

Comment:
I would drop 'experimental.istio.io' - and just use the name 'mesh'.

In Istio, it is a reserved keyword and we use it in our configs.
For K8S - we can create a Gateway named 'mesh' in istio-system, to make any validator happy, and document
that for Istio the gateway has the role of the sidecar.

Reply:
switched to `mesh`